### PR TITLE
feat: If cdn only updates IP, set cdn peers state to PeerStateLeave

### PR DESCRIPTION
**Pull request by gaius-qi**
## Description
- If cdn only updates IP, set cdn peers state to `PeerStateLeave`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jim3ma** [comment]
Fix #949

**Review by jim3ma — APPROVED** [review]
LGTM

**Review by jim3ma — DISMISSED** [review]
LGTM

File: scheduler/resource/cdn.go [hunk 1]
```diff
@@ -274,6 +284,44 @@ func cdnsToNetAddrs(cdns []*config.CDN) []dfnet.NetAddr {
...
+		}
+
+		if !found {
+			diff = append(diff, x)
+		}
+	}
+
+	return diff
+}
+
```
Comment (244372610) [inline scheduler/resource/cdn.go:324]: The two nested loops above can be combined into one

**Review by hyy0322 — APPROVED** [review]
LGTM

**Pull request merged**